doc/arch/adr-002-form-representation.md

```diff
@@ -0,0 +1,95 @@
+# ADR 2: JSON for form representation and storage
+
+## Context
+
+Bugle forms needs a data model to support these uses:
+- Users create forms with an arbitrary number of questions.
+- Question forms can have a variety of field types—text, checkbox, radio buttons
+  and dropdown boxes. It should be easy to add more response types later.
+- Users provide an unknown number of responses for each question form. Each
+  response maps to the structure and requirements of its question form.
+- Sufficiently fast reads for retrieving form data and collating all answers in
+  the responses.
+
+The application currently uses PostgreSQL for storing user data.
+
+With that in mind, some options include:
+- Representing form questions and answers in JSON
+  - using PostgreSQL's `json` type for the column having the form data.
+  - using PostgreSQL's `text` type for the column having the form data.
+  - using PostgreSQL's `jsonb` type as the column having the form data.
+  - storing form data in a separate document store
+- Representing and storing form questions and answers in relations
+- Storing the form questions directly as HTML, and storing the answers in a relation
```

Comment:
There's some missing context between the Context and the Decision:
- What were the drawbacks/tradeoffs that made relations an inferior choice compared to JSON? Where does it lack on the items mentioned above: Users can have many forms, forms can have many questions, questions can have many responses and sufficiently fast reads.
- Why not do `storing form data in a separate document store`?
- What does it mean to store questions directly as HTML?

We could get on a call sometime if you want to think about these together or share your thoughts if you've thought about it already.

Reply:
Kiran flagged a lot these concerns as well, I haven't yet got around to writing them down. I'll get on a call once I draft a proper section talking about all the tradeoffs.